lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -38,7 +38,10 @@ def authenticate_totp(code, options = {})
           drift = options[:drift] || self.class.allowed_otp_drift_seconds
           raise "authenticate_totp called with no otp_secret_key set" if totp_secret.nil?
           totp = ROTP::TOTP.new(totp_secret, digits: digits)
-          totp.verify_with_drift(code, drift)
+          new_timestamp = totp.verify_with_drift_and_prior(code, drift, totp_timestamp)
+          return false unless new_timestamp
+          self.totp_timestamp = new_timestamp
+          return true
```

Comment:
Since this is the last thing the method returns, the `return` word can be omitted.